### ISSUE TITLE
Running a breakdown on a terraform project with --terraform-parse-hcl does not show AWS TGW Attachment cost

### ISSUE THREAD

**Issue by schroedjan** [infracost/infracost#1647]
Running a breakdown scan with `--terraform-parse-hcl` flag seems to omit AWS transit gateway attachments.

```
module.example_module.aws_cloudwatch_log_group.this
 ├─ Data ingested                                                                       Monthly cost depends on usage: $0.63 per GB
 ├─ Archival Storage                                                                   Monthly cost depends on usage: $0.0324 per GB
 └─ Insights queries data scanned                                              Monthly cost depends on usage: $0.0063 per GB

module.example_module.aws_kms_key.this
 ├─ Customer master key                                                                                     1  months                         $1.00
 ├─ Requests                                                                                Monthly cost depends on usage: $0.03 per 10k requests
 ├─ ECC GenerateDataKeyPair requests                                       Monthly cost depends on usage: $0.10 per 10k requests
 └─ RSA GenerateDataKeyPair requests                                       Monthly cost depends on usage: $0.10 per 10k requests

 OVERALL TOTAL                                                                                                                                            $1.00
```

Alternatively running the breakdown **without the flag** (using local terraform plan) seems to work fine!

```
module.example_module.aws_cloudwatch_log_group.this
 ├─ Data ingested                                                                       Monthly cost depends on usage: $0.63 per GB
 ├─ Archival Storage                                                                   Monthly cost depends on usage: $0.0324 per GB
 └─ Insights queries data scanned                                              Monthly cost depends on usage: $0.0063 per GB

module.example_module.aws_kms_key.this
 ├─ Customer master key                                                                                     1  months                         $1.00
 ├─ Requests                                                                                Monthly cost depends on usage: $0.03 per 10k requests
 ├─ ECC GenerateDataKeyPair requests                                       Monthly cost depends on usage: $0.10 per 10k requests
 └─ RSA GenerateDataKeyPair requests                                       Monthly cost depends on usage: $0.10 per 10k requests

module.example_module.aws_ec2_transit_gateway_vpc_attachment.this[0]
 ├─ Transit gateway attachment                                                                        730  hours                         $43.80
 └─ Data processed                                                                      Monthly cost depends on usage: $0.02 per GB

 OVERALL TOTAL                                                                                                                                           $44.80
```

I could provide an example project, if needed.

**Comment by aliscott** [comment]
Thanks @schroedjan 🙏 If you have an example project that would be super helpful to track down this issue.

**Comment by schroedjan** [comment]
Hey @aliscott,
currently I am not able to reproduce the problem with a simple example project.
I need to further replicate our submodule structure and will get back to you once I can reproduce the problem!

**Comment by aliscott** [comment]
Thanks @schroedjan let us know if you're able to find a way to reproduce this.

**Comment by alikhajeh1** [comment]
@schroedjan and @frabar if you get the chance, can you please try to see if you can reproduce this with [v0.10](https://www.infracost.io/docs/guides/v0.10_migration/)? Unfortunately there's not much we can do if we can't reproduce it.

**Comment by schroedjan** [comment]
Hey  @aliscott , @alikhajeh1 ,
sorry to have kept you waiting.

I really am not able to reproduce the behavior using simple (inline) modules.

Maybe the error is based in using Terraform Modules saved to Terraform Cloud Registry?!
Using the hcl parsing option I also run into problems downloading remote terraform modules (see #1667).
Only after running `terraform init` locally I'm able to run infracost, resulting in missing resources like stated before.

Using v0.10.1 doesn't fix the behaviour but introduces new errors in some workspaces, where resources are showing in the cost estimation that are definitely not part of the terraform plan (and are not showing up with v0.9.24).

Maybe it would be better to wait for #1667 to be fixed. Do you have an estimate on when it should be closed?

**Comment by hugorut** [comment]
Hey @schroedjan, If possible I'd love to get some more information to try and help resolve this issue:

1. `module.example_module.aws_ec2_transit_gateway_vpc_attachment.this[0]` seems like it relies on a `count` attribute. Could you paste the actual line of code that has this in? Also, if this line relies on any other resource/variable values could you indicate what they are. e.g:

```
resource "aws_ec2_transit_gateway_vpc_attachment" "example" {
  count = var.enabled ? var.attachment_counts : 0 // I need this line
  subnet_ids         = [aws_subnet.example.id]
  transit_gateway_id = aws_ec2_transit_gateway.example.id
  vpc_id             = aws_vpc.example.id
}

variable "attachment_counts" { // I need this block and 
   type = number
   default = 3
}

---

module "example_module" {
    enabled = true // I need this attribute
    attachment_counts = 4 // I need this attribute
}
```
2.  Could you paste the additional resources seen in `v0.10.1`? Thanks

---

To answer your question regarding https://github.com/infracost/infracost/issues/1667. This is being actively worked on and we hope for a release next week. Thanks

**Comment by alikhajeh1** [comment]
Hey @schroedjan, any update on [Hugo's questions above](https://github.com/infracost/infracost/issues/1647#issuecomment-1144494550)? It's hard to progress this one unfortunately without more info

**Comment by alikhajeh1** [comment]
Going to close this issue as we've not heard back and we need more info to debug. Feel free to re-open the issue!
